src/main/java/org/threadly/litesockets/Client.java

```diff
@@ -3,557 +3,358 @@
-    stats.addWrite(size);
-    writeBufferSize.addAndGet(-size);
-  }
+  public ByteBuffer provideReadByteBuffer();
   
   /**
-   * This is used to get the current size of the unRead readBuffer.
-   * If this is greater then the maxBufferSize the client will be removed
-   * from the SocketExecuters reading operations.
+   * <p>This is used to get the current size of the readBuffers pending reads.</p>
    * 
    * @return the current size of the ReadBuffer.
    */
-  public int getReadBufferSize() {
-    return readBufferSize.get();
-  }
+  public int getReadBufferSize();
   
   /**
-   * This is used to get the current size of the unWriten writeBuffer.
-   * If this is greater then the maxBufferSize the client will stop accepting
-   * new writes (except for forced writes).
+   * <p>This is used to get the current size of the unWriten writeBuffer.</p>
    * 
    * @return the current size of the writeBuffer.
    */
-  public int getWriteBufferSize() {
-    return writeBufferSize.get();
-  }
+  public int getWriteBufferSize();
   
   /**
    * This is used to get the currently set max buffer size.
    * 
    * @return the current MaxBuffer size allowed.  The read and write buffer use this independently.
    */
-  public int getMaxBufferSize() {
-    return maxBufferSize;
-  }
+  public int getMaxBufferSize();
   
   /**
-   * This is used by the SocketExecuter to set the ThreadExecuter this client is supposed to use.
-   * The ThreadExecuter works in a single threaded manor to keep all the client operations in order.
-   * 
-   * This might need to be exposed to your package but in general You should not be assigning this or
-   * changing it.
-   *  
-   * @param sei the ThreadExecuter to use for this client.
-   */
-  protected void setThreadExecuter(SubmitterExecutorInterface sei) {
-    this.sei = sei;
-  }
-  
-  /**
-   * This returns the SubmitterExecutorInterface for this client.
-   * This executer works in a single threaded way so blocking this executer is very determinantal to the client flow.
-   * 
-   * @return the ThreadExecuter for this client.  
-   */
-  protected SubmitterExecutorInterface getThreadExecuter() {
-    return sei;
-  }
-
-  /**
-   * Sets this client to the specified SocketExecuter.  This happens when the client is added to the SocketExecuter.
-   * This method should not be overridden. 
+   * <p> This returns this clients {@link Executor}.  The client must have been added to the {@link SocketExecuterInterface} or 
+   * this will return null.</p>
    * 
+   * <p> Its worth noting that operations done on this {@link Executor} can/will block Read callbacks on the 
+   * client, but it does provide you the ability to execute things on the clients read thread</p>
    * 
-   * @param ce the SocketExecuter to set for this client.
+   * @return The {@link Executor} for the client.
    */
-  protected void setSocketExecuter(SocketExecuterBase ce) {
-    this.ce = ce;
-  }
+  public Executor getClientsThreadExecutor();
```

Comment:
If you wanted to be more _threadly_ you can return the SubmitterExecutorInterface.  But it's easy to convert an Executor into a SubmitterExecutor anyways, so it's not a big deal either way.